kubernetes/utils/create_from_yaml.py

```diff
@@ -14,60 +14,119 @@
 
 
 import re
-import sys
 from os import path
 
 import yaml
-from six import iteritems
 
 from kubernetes import client
 
 
-def create_from_yaml(k8s_client, yaml_file, verbose=False, **kwargs):
+def create_from_file(k8s_client, yaml_path, verbose=False, **kwargs):
     """
-    Perform an action from a yaml file. Pass True for verbose to
-    print confirmation information.
-    Input:
-    yaml_file: string. Contains the path to yaml file.
-    k8s_cline: an ApiClient object, initialized with the client args.
+    Perform an action from a yaml file.
+
+    :param k8s_client: an ApiClient object, initialized with the client args
+    :yaml_path: the path on the file system to the yaml file
+    :type yaml_path: str
+    :param verbose: print confimation information
+    :type verbose: bool
+
+    Available parameters for performing the subsequent action:
+    :param async_req:
+    :type async_req: bool
+    :param include_uninitialized: include partially initialized resources in
+    the response.
+    :type include_uninitialized: bool
+    :param str pretty: pretty print the output
+    :param str dry_run: When present, indicates that modifications should not
+    be persisted. An invalid or unrecognized dry_run directive will result in
+    an error response and no further processing of the request.
+    Valid values are: - All: all dry run stages will be processed
+    """
+    with open(path.abspath(yaml_path)) as f:
+        k8s_api = create_from_yaml(k8s_client, f.read(), verbose=verbose,
+                                   **kwargs)
+
+    return k8s_api
+
+
+def create_from_yaml(k8s_client, yaml_str, verbose=False, **kwargs):
```

Comment:
this function would be unnecessary if you add create_from stream as you could just add your yaml string to a io.StringIO() object

Reply:
That is true. I will send a fix later .

Reply:
a concern is also backward compatibility
for that this function should probably be kept with the old behavior of using file paths

Though I am no developer I don't know the rules this project has for this.
it seems the 9.0 release has not yet been cut so maybe one can still change it before release

Reply:
@juliantaylor the function create_from_yaml now accepts a file or a string, which will be then parsed.